miner/worker.go

```diff
@@ -159,10 +159,17 @@ func (miner *Miner) prepareWork(genParams *generateParams, witness bool) (*envir
 	// to parent+1 if the mutation is allowed.
 	timestamp := genParams.timestamp
 	if parent.Time >= timestamp {
-		if genParams.forceTime {
-			return nil, fmt.Errorf("invalid timestamp, parent %d given %d", parent.Time, timestamp)
+		// CHANGE(taiko): block.timestamp == parent.timestamp is allowed in Taiko protocol.
+		if !miner.chainConfig.Taiko {
+			if genParams.forceTime {
+				return nil, fmt.Errorf("invalid timestamp, parent %d given %d", parent.Time, timestamp)
+			}
+			timestamp = parent.Time + 1
+		} else {
+			if parent.Time > timestamp {
```

Comment:
Let's keep the original code here:

<img width="883" alt="image" src="https://github.com/user-attachments/assets/82aa3a78-6798-4dad-bd70-2d7c848172a5">

Reply:
This is how the `taiko` branch of taiko-geth is modified.